Assets/MRTK/SDK/Experimental/PulseShader/Scripts/SurfacePulseOnSelect.cs

```diff
@@ -0,0 +1,204 @@
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.SurfacePulse
+{
+	[AddComponentMenu("Scripts/MRTK/SDK/SurfacePulse")]
+	public class SurfacePulseOnSelect : MonoBehaviour, IMixedRealityPointerHandler
+	{
+		[Tooltip("Shader parameter name to drive the pulse radius")]
+		public string ParamName = "_Pulse_";
+
+		[Tooltip("Shader parameter name to set the pulse origin, in local space")]
+		public string OriginParamName = "_Pulse_Origin_";
+
+		[Tooltip("How long in seconds the pulse should animate")]
+		public float PulseDuration = 5f;
+
+		[Tooltip("How long to wait in seconds between pulses, when pulsing is active")]
+		public float PulseRepeatDelay = 5f;
+
+		[Tooltip("Minimum time to wait between each pulse")]
+		public float PulseRepeatMinDelay = 1f;
+
+		[Tooltip("Automatically begin repeated pulsing")]
+		public bool bAutoStart = false;
+
+		[Tooltip("Automatically set pulse origin to the main camera location")]
+		public bool bOriginFollowCamera = false;
+
+		[Tooltip("The material to animate")]
+		public Material SurfaceMat;
+
+		// Internal state
+		Coroutine RepeatPulseCoroutine;
+
+		float pulseStartedTime;
+		bool cancelPulse;
+
+
+		// Reset the material property when exiting play mode so it won't be changed on disk
+#if UNITY_EDITOR
+
+		SurfacePulseOnSelect()
+		{
+			EditorApplication.playModeStateChanged += HandleOnPlayModeChanged;
+		}
+
+		void HandleOnPlayModeChanged(PlayModeStateChange change)
+		{
+			// This method is run whenever the playmode state is changed.
+			if (!EditorApplication.isPlaying)
+			{
+				// do stuff when the editor is paused.
+				ResetPulseMaterial();
+			}
+		}
+
+#endif // UNITY_EDITOR
+
+		private void OnDestroy()
+		{
+			ResetPulseMaterial();
+			CoreServices.InputSystem?.UnregisterHandler<IMixedRealityPointerHandler>(this);
+
+		}
+
+		private void Start()
+		{
+			CoreServices.InputSystem?.RegisterHandler<IMixedRealityPointerHandler>(this);
```

Comment:
It feels odd to me that this needs to be a global listener but the `OnPointerClicked` assumes the pointer will have a hit point on the mesh. Can this instead be a focused listener on the parent object of the meshes? That way any mesh that gets tapped will have the event propagate up to the parent, and it can gather them all.

Reply:
I am not sure (I am not the original author : ) Is it possible to get the parent object of dynamically created spatial mesh and add a script?

Reply:
I believe `CoreServices.SpatialAwarenessSystem.SpatialAwarenessObjectParent` should give you the parent used for any meshes, right @davidkline-ms?

Reply:
Thanks @keveleigh @davidkline-ms ! I tried removing the global listener and adding PointerHandler to the parent object of the dynamically generated spatial mesh. Tested on the device and works well. Could you please review the code and see if this is correct use? 
For the example mesh object in-editor demo, I manually added Pointer Handler and used OnPointerClick event. 
![2020-08-25 22_41_13-](https://user-images.githubusercontent.com/13754172/91181919-7a9a6880-e724-11ea-9c74-48ee53dad183.png)